breathecode/admissions/views.py

```diff
@@ -788,19 +788,19 @@ def delete(self, request, cohort_id=None, academy_id=None):
         except Cohort.DoesNotExist:
             raise ValidationException("Cohort doesn't exist", code=400)
 
-        cohort.stage = DELETED
-        cohort.save()
-
         # Student
         cohort_users = CohortUser.objects.filter(
             role=STUDENT,
             cohort__id=cohort_id
         )
 
-        # TODO: this in one future maybe will be removed
-        for cohort_user in cohort_users:
-            cohort_user.delete()
+        # Check if cohort has students before deleting
+        if cohort_users.count() > 0:
```

Comment:
We need this validation also in the line 775, the line 775 cover one bulk delete over `academy/cohort`